scripts/extract-extension.js

```diff
@@ -0,0 +1,52 @@
+const JSZip = require('jszip');
+const { isValidExtensionName } = require('./lib/ExtensionNameValidator');
+const { createWriteStream } = require('fs');
+const { readFile } = require('fs/promises');
+const pipeline = require('util').promisify(require('stream').pipeline);
+
+/**
+ * Extracts exactly one extension into the community extensions folder from a zip file.
+ * @param {string} [zipPath] The path to the zip file to extract.
+ * @param {string} [extensionsFolder] The folder with the extensions.
+ * @returns {Promise<{error: "too-many-files" | "no-json-found"| "invalid-file-name" | "zip-error", details?: any} | {error?: undefined,extensionName: string}>} the name of the extracted extension if successful, else a generic error code.
+ */
+exports.extractExtension = async function (
+  zipPath = '/tmp/ext.zip',
```

Comment:
Don't make such an important parameter implicit:
- not setting it would most likely be an error from the caller (and confusing for the reader "who the heck do they know where the extension to extract is??")
- it's dividing the places where `/tmp/ext.zip` is set to two files, so there is one more chance one is updated without the other by a future ("why this file was not passed by the caller, argh I forgot to update it!!").